src/Features/Core/Portable/ConvertForToForEach/AbstractConvertForToForEachCodeRefactoringProvider.cs

```diff
@@ -100,7 +100,7 @@ internal abstract class AbstractConvertForToForEachCodeRefactoringProvider<
             //      dim i as integer
```

Comment:
why "VB allows"? what about c#:
```c#
int i;
for (i = 0; i < array.Length; ++i)
{
}
```
do we have tet for that? i didn't find any

Reply:
Because in VB it's the same syntactic form for each.  i.e. you can't tell the difference *syntactically* between:

```vb
for i = 0 to 10
next
```

and 

```vb
for i = 0 to 10
next
```

:)

In C# they are syntactically different, and we only run the transform on C# for the syntactic form that declares a variable in the for-loop.

Reply:
Added test for C# case just to ensure this works as expected and doesn't rever regress.  #resolved

Reply:
ok, i'd still like to have a test with `for (i = 0` in c# to make sure it's not offered and doesn't crash

Reply:
Yup.  Added TestWithInitializerOfVariableOutsideLoop for that purpose.
#resolved